donate/payments/tasks.py

```diff
@@ -62,11 +63,15 @@ def send_transaction_to_basket(data):
 class BraintreeWebhookProcessor:
 
     def process(self, notification):
-        process_method = getattr(self, 'process_{}'.format(notification.kind), None)
-        if process_method is not None:
-            return process_method(notification)
-
-    def process_subscription_charged_successfully(self, notification):
+        kind = notification.kind
+        if kind == 'subscription_charged_successfully':
+            self.subscription_charged_successfully(notification)
+        elif kind == 'subscription_charged_unsuccessfully':
+            self.subscription_charged_unsuccessfully(notification)
+        elif kind == 'dispute_lost':
+            self.dispute_lost(notification)
```

Comment:
this code was a bit _too_ magical. This rewrite changes things so that there are three explicit possible code paths, rather than an unknown number of potential failure paths.

Reply:
But then these are redefined in the actual methods below...why not just a comment that says "get the notification kind and call the class method with the matching name"?

Reply:
Because it really was _too_ magical. The refactor was trivial and hardens a part of the code that we needed to touch anyway.

Reply:
Ok, you care about this more than I do.

Reply:
I've got no problem using if/elif over `getattr`.